docs/gcloud-auth.rst

```diff
@@ -0,0 +1,208 @@
+.. toctree::
+  :maxdepth: 1
+  :hidden:
+
+Authentication
+--------------
+
+=======
+Summary
+=======
+
+For the majority of cases,
+authentication should "just work"
+as credentials can be loaded implicitly
+based on the environment you're application is running in
+and therefore there is no extra code required to authenticate.
```

Comment:
This is a tricky one. We can't cover *everything* under the sun, but I agree that these are serious potential pitfalls.

How about we add a FAQ at the end that has a few things:

1. How do I create a service account? (see links above)
1. I'm on GCE, but when I get an Access Denied message when trying to talk to Datastore (or Storage, or whatever). What gives?

(And I'll let you open a PR for more...)